<div class="alert alert-block alert-success">
    <h1 align="center">Machine Learning in Python</h1>
    <h3 align="center">Behrang Mz.</h3>
</div>

<img src = "https://www.cyclonis.com/images/2020/03/googleplay.jpg" width=50%>

## Importing the libraries

In [1]:
import matplotlib
import pandas as pd

## Load and Prepare Data

In [2]:
gp_data = pd.read_csv('googleplaystore.csv')
gp_data.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [3]:
gp_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


<div dir="rtl" class="alert alert-block alert-info">
    <h1 align="center">اکتشاف داده ها</h1>
</div>

In [63]:
gp_data.Rating.isnull().value_counts(normalize=True)*100

False    86.403468
True     13.596532
Name: Rating, dtype: float64

<div dir="rtl">
 این دیتاست شامل 10841 سطر داده است. از این میان تنها داده های ستون Rating به صورت عددی هستند و بقیه ستون ها نوع داده ای String دارند.
    <br>
    بیش از 13درصد از داده های موجود در ستون Rating بدون مقدار هستند! 
    <br>
    در ستون Type و Content Rating یک داده نامعتبر یا missing data قرار دارد
    <br>
    ستون های App و Current Ver و Android Ver و Last Updated نیز شامل داده هایی از جنس String بوده و برای Machine Learning به کار نمی آیند.
</div>

In [69]:
gp_data[gp_data['Type'].isnull()]

App Category  Rating Reviews                Size  \
9148  Command & Conquer: Rivals   FAMILY     NaN       0  Varies with device   

     Installs Type Price Content Rating    Genres   Last Updated  \
9148        0  NaN     0   Everyone 10+  Strategy  June 28, 2018   

             Current Ver         Android Ver  
9148  Varies with device  Varies with device

In [68]:
gp_data[gp_data['Content Rating'].isnull()]

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

<div dir="rtl">
    داده های موجود در ستون Content Rating شامل مقادیر غیرتکراری زیر است:
</div>

In [4]:
print(gp_data['Content Rating'].unique())

['Everyone' 'Teen' 'Everyone 10+' 'Mature 17+' 'Adults only 18+' 'Unrated'
 nan]


## EDA (Use Pandas)

<div dir="rtl" class="alert alert-block alert-success">
    <h2 align="right">علاقمندی گروه های سنی مختلف به کدام گروه از اپلیکیشن ها بیشتر بوده است (سه دسته با بیشترین محبوبیت)؟</h2>
</div>

In [7]:
sen1 = pd.crosstab(gp_data.Category, gp_data['Content Rating'], normalize='columns')[['Everyone 10+', 'Teen', 'Mature 17+', 'Adults only 18+']]

In [12]:
evro = sen1.sort_values('Everyone 10+', ascending=False)[['Everyone 10+']][:3]*100
teen = sen1.sort_values('Teen', ascending=False)[['Teen']][:3]*100
matr = sen1.sort_values('Mature 17+', ascending=False)[['Mature 17+']][:3]*100
adlt = sen1.sort_values('Adults only 18+', ascending=False)[['Adults only 18+']][:3]*100

In [13]:
pd.concat([evro, teen, matr, adlt], axis=1).fillna('-')

Content Rating     Everyone 10+       Teen Mature 17+ Adults only 18+
Category                                                             
GAME                  31.642512  27.400662  14.829659               -
FAMILY                31.642512   21.60596          -               -
NEWS_AND_MAGAZINES    15.942029          -          -               -
SOCIAL                        -  10.347682  13.426854               -
DATING                        -          -  40.280561               -
COMICS                        -          -          -       66.666667
SPORTS                        -          -          -       33.333333
ART_AND_DESIGN                -          -          -             0.0

## Send Me the Result (on Telegram)